wf-galleon/WFLY-19021-Stability_In_Provisioning.adoc

```diff
@@ -0,0 +1,313 @@
+* `default` features + `experimental` and `preview` and `community` and `default` packages of C are provisioned
+
+If the package stability level `experimental` is specified:
+
+* `default` features/packages of A are provisioned
+* `community` and `default` features/packages of B are provisioned
+* `experimental`, `preview`, `community` and `default` features/packages of C are provisioned
+
+Example of a feature-pack containing packages and features at a lower stability level than its default stability levels:
+
+* The feature-pack A contains features at `default`, `community` and `preview` level.
+* The feature-pack A contains packages at `default`, `community` and `preview` level.
+* The default package stability level is `community`.
+* The default community stability level is `community`.
+* The minimum stability level is set to `preview`.
+
+If no stability level is specified:
+
+* `default` and `community` features/packages of A are provisioned
+
+If the package and config stability level `preview` is specified:
+
+* `default`, `community` and `preview` features/packages of A are provisioned
+
+==== Lowest config stability level needed by WildFly Galleon Plugins
+
+At provisioning time, when WildFly Galleon plugin generates the server configurations, 
+a WildFly embedded server is started. This server is started with a stability level computed in the following way:
+
+* If a config stability level has been specified by the user, this level is used.
+* If no config stability level has been specified, the lowest config stability level of all feature-packs present in the provisioning is used.
+
+=== Provisioning tooling requirement
+
+The following Galleon based tooling should let users specify the `config-stability-level` and `package-stability-level` options:
+
+* Galleon CLI and Maven plugin
+* WildFly Maven Plugin
+* WildFly Bootable JAR Maven Plugin
+* WildFly Glow
+
+* In addition, for WildFly Glow, information about the features/packages that would be not provisioned for a given stability level 
+is displayed to the user.
+
+=== Nice-to-Have Requirements
+
+* Support for stability in Prospero provisioning tooling.
+
+=== Non-Requirements
+
+* Recording and utilizing stability information in Galleon configuration elements 
+at a higher granularity level than a feature-spec or package (i.e. configs, layers, feature-groups).
+
+* If a provisioning plan (i.e. a provisioning.xml or other provisioning tooling configuration that instructs the tooling about what to provision) 
+involves configs, layers of feature-groups that reference features or packages that are not available at the stability level in effect, 
+the result is not specified. That is:
+** Always rejecting provisioning such a server is not required.
+** Always provisioning a working server is not required.
+** Best practice for feature pack authors is to not include features or packages at a lower stability level in configs, 
```

Comment:
@yersan , it is currently not supported.  We could make it strict. If we detect that a referenced package/feature is unknown, we could fail at feature-pack build time. @bstansberry would you be ok that we make it strict as a future enhancement of this proposal?

Reply:
@jfdenise Sure. Any 'future work' section is just suggestions anyway, not a commitment, so if you think it's a reasonable thing to do, go for it.

BTW I suggested the language here about not guaranteeing any behavior, but that was because I wanted to limit the immediate scope for this iteration while also being clear about the topic in the analysis (i.e. don't let readers assume anything.) I don't have any particular feelings about what should be done later.

That said, AIUI WFGP-278 isn't really about what I was thinking about for this. My impression there is there's some optional module.xml dependency that is resulting in a provisioning failure, and I'd expect that to just work. But we should discuss the details of that particular case elsewhere.

Reply:
I added a section.